docs/jwst/extract_2d/main.rst

```diff
@@ -78,6 +78,80 @@ of ``GrismObjects`` outside of these, the ``GrismObject`` itself can be imported
 The dispersion direction will be documented by copying keyword "DISPAXIS"
 (1 = horizontal, 2 = vertical) from the input file to the output cutout.
 
+Following are examples of how to customize the list of grism objects used in extract_2d.
+The input file must have a WCS assigned to it by running ``assign_wcs``. The default values
+for  wavelength range and the spectral orders are stored in the ``wavelengthrange``
+reference file which can be retrieved from CRDS. A user may supply a different
+wavelength range by passing `None` to ``reference_files``. In this case the spectral
+orders to be extracted and their corresponding wavelength range will be taken
+from the ``wavelength_range`` parameter.
+
+.. doctest-skip::
+
+  >>> from jwst.datamodels import ImageModel
+  >>> input_model = ImageModel("nircam_wfss_assign_wcs.fits")
+
+Retrieve the wavelengthrange file specific for this mode:
+
+.. doctest-skip::
+
+  >>> from jwst.extract_2d import Extract2dStep
+  >>> step = Extract2dStep()
+  >>> refs = {}
+  >>> for ref_type in step.reference_file_types:
+  ...     refs[ref_type] = step.get_reference_file(input_model, ref_type)
+  >>> print(refs)
+  {'wavelengthrange': '/crds/references/jwst/niriss/jwst_niriss_wavelengthrange_0002.asdf'}
+
+Create a list of grism objects for a specified spectral order with a limited
+minimum magnitude, and a specified half height of the extraction box in
```

Comment:
Should note that the user-specified extraction height is only applied to grism objects with a source type of POINT (non-point sources should use the normal bb limits).